.githooks/pre-commit

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Exit on error
+# exit when any command fails
+set -e
+set -o pipefail
+
+# Credit for this great script goes to Gustav Wengel at https://github.com/GeeWee/reshaper-pre-commit-hook
+# This has been modified for use in this project and to use dotnet-format instead.
+
+# Copyright 2020 Gustav Wengel
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+# TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+# CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+# Folder with githooks files
+GIT_HOOKS_DIR="./.githooks"
+
+# keep track of the last executed command
+trap 'last_command=$current_command; current_command=$BASH_COMMAND' DEBUG
+# echo an error message before exiting
+trap 'echo "\"${last_command}\" command filed with exit code $?."' ERR
+
+# - Actual script
+STAGED_FILES=`git diff --name-only --cached --diff-filter=d`
+echo "Staged files ${STAGED_FILES}"
+
+# Build edit string, by replacing newlines with semicolons.
+# --diff-filter=d only filters files that are not deleted, which means we won't have trouble adding them afterwards
+INCLUDE_STRING=`git diff --name-only --cached --diff-filter=d | sed ':a;N;$!ba;s/\n/;/g'`
+echo "Include string: $INCLUDE_STRING"
+
+# If the include string is empty, we're done. This happens e.g. if the commit only consists of deleted files.
+if [[ -z "$INCLUDE_STRING" ]]
+then
+    echo "No files to change"
+else
+    # Edit your project files here
+    echo "Formatting files..."
+    dotnet tool restore # This handles cache. If there's no restore needed it will not download anything.
```

Comment:
Is dotnet already in the install instructions? If not we should add it to ensure users have to tool installed.

Reply:
We can add this update in another PR, just have to verify and will make a new issue if it's not already in the instructions.

Reply:
Resolved, included as part of the installing `.NET Core SDK`